### PR TITLE
Added namespace descriptions.

### PR DESCRIPTION
**Pull request by rpetrusha**
Table containing namespaces and namespace descriptions.

/cc @blackdwarf @mairaw @stevehoag

### CODE REVIEW

File: api/index.md [hunk 1]
```diff
@@ -5,13 +5,570 @@ _navPath: toc.html
 _navRel: ../toc.html
 ---
 
-# .NET Core API Reference
+# .NET Core API Reference 
 
-These documents cover the API reference for .NET Core. You can use the search function 
-on the left-hand side to search for an API or use the Table of Contents on the left to navigate.
+The .NET Core API Reference documents the types and type members available in .NET Core. You can navigate to the documentation for a particular type (a class, structure, delegate, enum, or interface) or type member (a constructor, method, property, or event) in any of three ways:
```
Comment (svick) [inline api/index.md]: If you're enumerating all the possible members, why do you leave out fields? Sure, fields are rare in public APIs, but [they do exist](http://dotnet.github.io/api/System.String.html#fields).

File: api/index.md [hunk 2]
```diff
@@ -5,13 +5,570 @@ _navPath: toc.html
...
+   <td>Provides types that support interoperation between the Dynamic Language Runtime and C#.
+   </td>
+</tr>
+<tr>
+   <td><a href="http://dotnet.github.io/api/Microsoft.SqlServer.Server.html" > Microsoft.SqlServer.Server</a></td>
+   <td>Contains classes for integrating .NET Core with Microsoft sQL Server and the SQL Server database engine process execution environment.
+   </td>
+</tr>
+<tr>
+   <td><a href="http://dotnet.github.io/api/Microsoft.VisualBasic.html">Microsoft VisualBasic</a></td>
```
Comment (svick) [inline api/index.md]: Space instead of `.`: Microsoft VisualBasic → Microsoft.VisualBasic

File: api/index.md [hunk 3]
```diff
@@ -5,13 +5,570 @@ _navPath: toc.html
...
-These documents cover the API reference for .NET Core. You can use the search function 
-on the left-hand side to search for an API or use the Table of Contents on the left to navigate.
+The .NET Core API Reference documents the types and type members available in .NET Core. You can navigate to the documentation for a particular type (a class, structure, delegate, enum, or interface) or type member (a constructor, method, property, or event) in any of three ways:
 
-If you find an issue or an inconsistency, we would love to hear about it! 
-Go ahead to [.NET Core docs repo](https://github.com/dotnet/core-docs/) On Github and leave 
-an issue to let us know.
+- By using the search function in the left-hand pane to search for the **type** by name. (Search by member name is not supported.)
 
+- By navigating to the documentation for the type by expanding the **namespace** that that type is found in in the left hand pane, and then selecting the type.
```
Comment (mairaw) [inline api/index.md]: I'd rewrite this sentence to remove the duplicates that that and in in. Maybe **namespace** where that type is found in the left hand pane?

File: api/index.md [hunk 4]
```diff
@@ -5,13 +5,570 @@ _navPath: toc.html
...
+
+- By selecting the namespace in the "Namespaces in the .NET Core Class Library" section below.  
+ 
+Each topic contains the documentation for a type and all its unique public and protected members. Inherited members are not included. To view the inherited members of a type, click on one or more of its base classes in the Inheritance section immediately after the type description. 
+
+## Namespaces ##
+
+The types in the .NET Core class library documentation are organized by namespace. A namespace is a virtual label that allows a type or type member to be uniquely identified without requiring that that type name itself be unique. For example, a class library might include two types named `People` that have different members and offer different functionality. Locating them within namespaces (such as `Application.Collections.People` and `Application.ApplicationObjects.People`) allows compilers to resolve what might otherwise be ambiguous references to `People` objects.
```
Comment (mairaw) [inline api/index.md]: without requiring that that type name itself be unique ->
without requiring the type name itself to be unique

File: api/index.md [hunk 5]
```diff
@@ -5,13 +5,570 @@ _navPath: toc.html
...
+   </tr>
+</thead>
+<tr>
+   <td><a href="http://dotnet.github.io/api/Microsoft.CSharp.RuntimeBinder.html" >Microsoft.CSharp.RuntimeBinder</a></td>
+   <td>Provides types that support interoperation between the Dynamic Language Runtime and C#.
+   </td>
+</tr>
+<tr>
+   <td><a href="http://dotnet.github.io/api/Microsoft.SqlServer.Server.html" > Microsoft.SqlServer.Server</a></td>
+   <td>Contains classes for integrating .NET Core with Microsoft sQL Server and the SQL Server database engine process execution environment.
```
Comment (mairaw) [inline api/index.md]: sQL -> SQL

File: api/index.md [hunk 6]
```diff
@@ -5,13 +5,570 @@ _navPath: toc.html
...
+<tr>
+   <td><a href="http://dotnet.github.io/api/System.Collections.ObjectModel.html">System.Collections.ObjectModel</a></td>
+   <td>
+   Contains classes that can be used as collections in the object model of a reusable library. Use these classes when properties or methods return collections.
+   </td>
+</tr>
+<tr>
+   <td><a href="http://dotnet.github.io/api/System.Collections.Specialized.html">System.Collections.Specialized</a></td>
+   <td>
+   Contains specialized and strongly-typed collectionsm such as a linked list dictionary, a bit vector, and collections that contain only strings.
```
Comment (mairaw) [inline api/index.md]: typo: collectionsm

File: api/index.md [hunk 7]
```diff
@@ -5,13 +5,570 @@ _navPath: toc.html
...
+<tr>
+   <td><a href="http://dotnet.github.io/api/System.Reflection.Metadata.html">System.Reflection.Metadata</a></td>
+   <td>
+   Contains a class with single extension method that retrieves raw metadata from an assembly. 
+   </td>
+</tr>
+<tr>
+   <td><a href="http://dotnet.github.io/api/System.Resources.html">System.Resources</a></td>
+   <td>
+   Contains classes that allow developers to create, store, and retreive culture-specific string resources used in an application. One of the most important classes of the <a href="http://dotnet.github.io/api/System.Resources.html">System.Resources</a> namespace is the <a href="http://dotnet.github.io/api/System.Resources.ResourceManager.html">ResourceManager</a> class.
```
Comment (mairaw) [inline api/index.md]: retreive -> retrieve

File: api/index.md [hunk 8]
```diff
@@ -5,13 +5,570 @@ _navPath: toc.html
...
+<tr>
+   <td><a href="http://dotnet.github.io/api/System.Text.RegularExpressions.html">System.Text.RegularExpressions</a></td>
+   <td>
+   Contains classes that represent the .NET regular expression engine, which allows you to search text for particular patterns. 
+   </td>
+</tr>
+<tr>
+   <td><a href="http://dotnet.github.io/api/System.Threading.html">System.Threading</a></td>
+   <td>
+   Provides types that enable multithreaded programming. In addition to classes for synchronizing thread activities and access to data (`Mutex`, `Monitor`, `Interlocked`, `AutoResetEvent`, and so on), this namespace includes a `ThreadPool` class that allows you to use a pool of system-supplied threads, and a `Timer` class that executes callback methods on thread pool threads.
```
Comment (mairaw) [inline api/index.md]: in some cases you're linking to the class, in others formatting as code. I think it would be good to be consistent

**Comment by svick** [comment]
Should the documentation contain implicit suggestions that using old namespaces is recommended?

For example:

> System.Collections.Concurrent
> Provides thread-safe collection classes that should be used instead of the corresponding types in the **System.Collections** and System.Collections.Generic namespaces whenever multiple threads are accessing the collection concurrently.

To me, this implies that using non-generic collections from System.Collections is fine when you don't need thread safety.

Or:

> System.Collections.ObjectModel
> Contains classes that can be used as collections in the object model of a reusable library. **Use these classes when properties or methods return collections.**

To me, this sounds like it's saying that returning `Collection<T>` is better than returning `IList<T>` or `List<T>`.

File: api/index.md [hunk 9]
```diff
@@ -5,13 +5,570 @@ _navPath: toc.html
...
+<tr>
+   <td><a href="http://dotnet.github.io/api/System.Data.Common.html">System.Data.Common</a></td>
+   <td>
+   Contains classes shared by the .NET Core data providers.
+   </td>
+</tr>
+<tr>
+   <td><a href="http://dotnet.github.io/api/System.Data.SqlClient.html">System.Data.SqlClient</a></td>
+   <td>
+   The data provider for SQL Server.
```
Comment (svick) [inline api/index.md]: Maybe say "Microsoft SQL Server", to make it clear this is not for generic SQL servers?

File: api/index.md [hunk 10]
```diff
@@ -5,13 +5,570 @@ _navPath: toc.html
...
+<tr>
+   <td><a href="http://dotnet.github.io/api/System.Diagnostics.Contracts.html">System.Diagnostics.Contracts</a></td>
+   <td>
+   Contains static classes for representing program contracts such as preconditions, postconditions, and invariants.
+   </td>
+</tr>
+<tr>
+   <td><a href="http://dotnet.github.io/api/System.Diagnostics.Tracing.html">System.Diagnostics.Tracing</a></td>
+   <td>
+   Provides types and members that enable you to create strongly-typed events to be captured by event tracing for Windows (ETW).
```
Comment (svick) [inline api/index.md]: Shouldn't this be capitalized? (Event Tracing for Windows)

File: api/index.md [hunk 11]
```diff
@@ -5,13 +5,570 @@ _navPath: toc.html
...
+<tr>
+   <td><a href="http://dotnet.github.io/api/System.Linq.Expressions.html">System.Linq.Expressions</a></td>
+   <td>
+   Contains types that enable language-level code expressions to be represented as objects in the form of expression trees.
+   </td>
+</tr>
+<tr>
+   <td><a href="http://dotnet.github.io/api/System.Net.html">System.Net</a></td>
+   <td>
+   Provides a simple programming interface for many of the protocols used on networks today. The <a href="http://dotnet.github.io/api/System.Net.WebRequest.html">WebRequest</a> and <a href="http://dotnet.github.io/api/System.Net.WebResponse.html">WebResponse</a> classes form the basis of what are called pluggable protocols, an implementation of network services that enables you to develop applications that use Internet resources without worrying about the specific details of the individual protocols.
```
Comment (svick) [inline api/index.md]: I'm not sure it makes sense to focus on pluggability here. I don't think that feature is used much, and I think `WebRequest` exists in .Net Core pretty much only for compatibility.

File: api/index.md [hunk 12]
```diff
@@ -5,13 +5,570 @@ _navPath: toc.html
...
+<tr>
+   <td><a href="http://dotnet.github.io/api/System.Net.Security.html">System.Net.Security</a></td>
+   <td>
+   Provides network streams for secure communications between hosts.
+   </td>
+</tr>
+<tr>
+   <td><a href="http://dotnet.github.io/api/System.Net.Sockets.html">System.Net.Sockets</a></td>
+   <td>
+   Provides a managed implementation of the Windows Sockets (Winsock) interface for developers who need to tightly control access to the network.
```
Comment (svick) [inline api/index.md]: I'm not sure it makes sense to talk about Winsock, since .Net Core is cross-platform.

File: api/index.md (hunk 7 above)
Comment (svick) [inline api/index.md]: What's the point to linking to System.Resources again in the description?

File: api/index.md [hunk 13]
```diff
@@ -5,13 +5,570 @@ _navPath: toc.html
...
+<tr>
+   <td><a href="http://dotnet.github.io/api/System.Runtime.ExceptionServices.html">System.Runtime.ExceptionServices</a></td>
+   <td>
+   Contains a single class to represent an exception whose state is captured at a certain point in code. 
+   </td>
+</tr>
+<tr>
+   <td><a href="http://dotnet.github.io/api/System.Runtime.InteropServices.html">System.Runtime.InteropServices</a></td>
+   <td>
+   Provides a wide variety of members that support COM interop and platform invoke services on Windows platforms.
```
Comment (svick) [inline api/index.md]: I think InteropServices types are used for PInvoke on Unix too, not just on Windows.

File: api/index.md [hunk 14]
```diff
@@ -5,13 +5,570 @@ _navPath: toc.html
...
+<tr>
+   <td><a href="http://dotnet.github.io/api/System.Security.Cryptography.X509Certificates.html">System.Security.Cryptography.X509Certificates</a></td>
+   <td>
+   Contains the runtime's implementation of the Authenticode X.509 v.3 certificate. This certificate is signed with a private key that uniquely and positively identifies the holder of the certificate.
+   </td>
+</tr>
+<tr>
+   <td><a href="http://dotnet.github.io/api/System.Security.Cryptography.Xml.html">System.Security.Cryptography.Xml</a></td>
+   <td>
+   Contains a single type that represents the **<X509IssuerSerial>** element of an XML digital signature.
```
Comment (svick) [inline api/index.md]: The XML tag is not visible.

File: api/index.md [hunk 15]
```diff
@@ -5,13 +5,570 @@ _navPath: toc.html
...
+<tr>
+   <td><a href="http://dotnet.github.io/api/System.Xml.XPath.html">System.Xml.XPath</a></td>
+   <td>
+   Contains types that define a cursor model for navigating and editing XML information items as instances of the XQuery 1.0 and XPath 2.0 Data Model.
+   </td>
+</tr>
+<tr>
+   <td><a href="http://dotnet.github.io/api/System.Xml.Xsl.html">System.Xml.Xsl</a></td>
+   <td>
+   Provides support for Extensible Stylesheet Transformation (XSLT) transforms. It supports the <a href="http://www.w3.org/TR/xslt">W3C XSL Transformations (XSLT) Version 1.0 Recommendation</a>.
```
Comment (svick) [inline api/index.md]: Why is there a link for XSLT here, but not for XSD, XQuery and XPath above?

**Comment by svick** [comment]
I noticed that this was added to master (3809737). Should this PR be closed?

**Comment by richlander** [comment]
@rpetrusha ?

**Comment by mairaw** [comment]
Good question @svick. Normally I'd expect to see the commit in this same PR for us to review the changes and merge.

**Pull request closed without merging**